pkg/transaction/transaction.go

```diff
@@ -578,3 +581,11 @@ func (t *transactionService) Close() error {
 	t.wg.Wait()
 	return nil
 }
+
+func (t *transactionService) TransactionFee(ctx context.Context, txHash common.Hash) (*big.Int, error) {
+	hash, _, err := t.backend.TransactionByHash(ctx, txHash)
+	if err != nil {
+		return nil, err
+	}
+	return hash.GasPrice().Mul(hash.GasPrice(), big.NewInt(int64(hash.Gas()))), nil
```

Comment:
as it turns out, the transaction type returned by `TransactionByHash` has a function called `Cost()` which calculates what you do on line 590 for you :)

Reply:
Thanks God.